File: DOSSOU_Yawavi_Flore/backend/src/main/java/com/maxime/Ega/entity/Transaction.java
package com.maxime.Ega.entity;

import com.maxime.Ega.enums.TransactionType;
import jakarta.persistence.*;
import jakarta.validation.constraints.PositiveOrZero;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "transactions")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Transaction {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "transaction_id",  nullable = false)
    private String transactionId;

    @Column(name = "transaction_type", nullable = false)
    @Enumerated(EnumType.STRING)
    private TransactionType transactionType;

    @Column(name = "transaction_date", nullable = false)
    private LocalDateTime transactionDate;

    @Column(name = "amount", nullable = false)
    @PositiveOrZero
    private BigDecimal amount;

    @Column(name = "description")
    private String description;

    @ManyToOne
    private Account account;

    @ManyToOne
    private User user;
}

File: Ega_Frontend/src/app/models/Transaction.ts
import Decimal from "decimal.js";
import { TransactionType } from "../enums/TransactionType";
import { Account } from "./Account";
import { User } from "./User";

export class Transaction{
    transactionId? : string ;
    transactionType? : TransactionType ;
    amount? : Decimal ;
    account?: Account ;
    user? : User;

    constructor(data?: Partial<Transaction>){
        if(data){
            Object.assign(this, data)
        }
    }
}